src/components/modal/ModalAddEdit.vue

```diff
@@ -107,13 +97,13 @@ export default {
       };
 
       if (this.productData.data !== undefined) {
-        Http.put("/api/products/" + this.productData.data.id, data);
+        Http.put("products/" + this.productData.data.id, data);
       } else {
-        Http.post("/api/products/", data);
+        Http.post("products/", data);
       }
 
       this.hideModalHandler();
-      this.fetchProductData();
+      await this.$store.dispatch("setProductList");
```

Comment:
Nitpick: If you don't need to execute anything after this method it's not necessary to `await` for it. 😉   
Also, consider using `mapActions` to reference vuex parts instead of directly accessing it. 😉